### PR TITLE
Remove dependency on Internal.AspNetCore.BuildTasks

### LINKED ISSUE #32492: Arpow: GenerateFileFromTemplate task causes BuildTools prebuilt
### Describe the bug
The `GenerateFileFromTemplate` task is defined in BuildTools, which isn't built with source-build.  This will be reported as a prebuilt in Arpow when building offline and when prebuilt reporting is turned on.  This task should either be moved to Arcade or moved into RepoTasks to ensure it can build and be used with source-build.

### To Reproduce
When Arpow source-build CI is included (see https://github.com/dotnet/aspnetcore/issues/31445), in the generated results (`Microsoft.SourceBuild.Intermediate.aspnetcore.*.nupkg`), the file `prebuilt-report/annotated-usage.xml` will contain a reference to BuildTools, indicating that this is a prebuilt.

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Fixes: #32492 

Instead, use the Microsoft.DotNet.Build.Tasks.Templating package from dotnet/arcade https://github.com/dotnet/arcade/pull/7403.

### CODE REVIEW

File: eng/tools/GenerateFiles/GenerateFiles.csproj [hunk 1]
```diff
@@ -4,6 +4,10 @@
     <TargetFramework>net5.0</TargetFramework>
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="Microsoft.DotNet.Build.Tasks.Templating" PrivateAssets="All" Version="$(MicrosoftDotNetBuildTasksTemplatingVersion)" IsImplicitlyDefined="true" />
```
Comment (dougbu) [inline eng/tools/GenerateFiles/GenerateFiles.csproj:8]: Like referencing the package only where we need it😺

File: src/ProjectTemplates/GenerateContent.targets [hunk 2]
```diff
@@ -47,8 +51,8 @@
       Properties="$(GeneratedContentProperties);%(GeneratedContent.AdditionalProperties)"
       OutputPath="%(GeneratedContent.OutputPath)">
 
-      <Output TaskParameter="OutputPath" ItemName="FileWrites" />
-      <Output TaskParameter="OutputPath" ItemName="Content" />
+      <Output TaskParameter="ResolvedOutputPath" ItemName="FileWrites" />
+      <Output TaskParameter="ResolvedOutputPath" ItemName="Content" />
```
Comment (dougbu) [inline src/ProjectTemplates/GenerateContent.targets:55]: What's this for❔

**Review by dougbu — COMMENTED** [review]
I'm about ready to approve but would appreciate an answer&hellip;

File: src/ProjectTemplates/GenerateContent.targets (hunk 2 above)
Comment (JunTaoLuo) [reply src/ProjectTemplates/GenerateContent.targets:55]: https://github.com/dotnet/arcade/pull/7403#discussion_r645164152

**Review by dougbu — APPROVED** [review]

**Pull request merged**